lib/mongoose_push/api/v3/h2_handler.ex

```diff
@@ -0,0 +1,52 @@
+defmodule MongoosePush.API.V3.H2Handler do
+  @moduledoc """
+  Module for handling internal responses to V3 HTTP2 codes
+  """
+  @behaviour MongoosePush.API
+  alias MongoosePush.Service
+
+  @spec to_status(:ok | {:error, Service.error()} | {:error, MongoosePush.error()}) ::
+    {non_neg_integer, %{details: atom | String.t()} | nil}
+  def to_status(:ok), do: {200, nil}
+
+  def to_status({:error, {type, reason}}) when is_atom(reason) do
+    return_code =
+      case type do
+        :invalid_request ->
+          461
+        :auth ->
+          462
+        :unregistered ->
+          463
+        :unspecified ->
+          464
+        :service_internal ->
```

Comment:
`service_internal`, `auth`, both should be 5xx, as they are not in any way connected to the user's input. The auth one is all about auth issues between MongoosePush and Push Service, not user-input auth.

Reply:
`500` for `service_internal` and `511` for `auth` then ?

Reply:
Both would be very misleading to client. 
`500` means that it's the server client is connecting to misbehaving which is not true `service_internal` means that 3rd party service APNS / FCM has issues. 
`511` is some weird version of `401`used for network / proxy services. Either way, it suggest that client is missing authorization which is not true at all - `auth` is an error category that all mean that MongoosePush itself is unable to authorize with APNS / FCM. This is normally a configuration error. 

There are no official HTTP codes that cover those cases. In general, what we need to do is to either pick the closest that makes sense (`503` here for both cases) and return well defined details in payload, or go with custom ones. 

Initially for MongoosePush, I figured that the amount of possible errors may be better of with being displayed as custom HTTP codes, but now that I think about it, we can scrap this idea. Thus, feel free to change all those codes to standard codes that correspond closely to our error reasons, group them and let's return well defined `%{reason => "internal_error_code"}` JSON payload.